### PR TITLE
Update raft engine

### PR DESCRIPTION
**Pull request by tabokie**
Signed-off-by: tabokie <xy.tao@outlook.com>

### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

Mainly to include metrics and interfaces: tikv/raft-engine#82, tikv/raft-engine#76.

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raft_log_engine/src/engine.rs [hunk 1]
```diff
@@ -173,25 +173,17 @@ impl RaftEngine for RaftLogEngine {
...
-            miss: stat.miss,
-            cache_size: stat.cache_size,
-        })
+        None
     }
 
-    fn stop(&self) {
-        self.0.stop();
-    }
+    fn stop(&self) {}
```
Comment (MrCroxx) [inline components/raft_log_engine/src/engine.rs:186]: Shall we support stop RaftLogEngine in future?

**Review by MrCroxx — APPROVED** [review]

File: components/raft_log_engine/src/engine.rs (hunk 1 above)
Comment (tabokie) [reply components/raft_log_engine/src/engine.rs:186]: Raft engine itself doesn't manage any threads anymore, so stop is simply exiting the client process.

**Review by Connor1996 — APPROVED** [review]
LGTM

**Review by hicqu — APPROVED** [review]

**Pull request merged**